consensus/istanbul/backend/api.go

```diff
@@ -89,6 +90,16 @@ func (api *API) GetValidators(number *rpc.BlockNumber) ([]common.Address, error)
 	return istanbul.MapValidatorsToAddresses(validators), nil
 }
 
+// GetValidatorsPublicKeys retrieves the list of validators public keys that must sign a given block.
+func (api *API) GetValidatorsPublicKeys(number *rpc.BlockNumber) ([]blscrypto.SerializedPublicKey, error) {
+	header, err := api.getParentHeaderByNumber(number)
+	if err != nil {
+		return nil, err
+	}
+	validators := api.istanbul.GetValidators(header.Number, header.Hash())
```

Comment:
Is this an expensive operation? In particular if asked for an old epoch. My "angle" here is.. would this provide a simple way to DoS the node?

Reply:
We already have a `GetValidators` operation that gets you the addresses, this shouldn't be more expensive than that.

Reply:
fair enough :P